src/widgets/radio.jsx

```diff
@@ -49,79 +189,54 @@ var BaseRadio = React.createClass({
+                                    this.props.multipleSelect ||
+                                    this.props.deselectEnabled;
+                                this.checkOption(
+                                    i,
+                                    shouldToggle ? !choice.checked : true);
                             }
-                            onClick={!this.props.labelWrap ? null : (e) => {
-                                // Don't send this to the scratchpad
-                                e.preventDefault();
-                                if (!this.props.apiOptions.readOnly) {
-                                    var shouldToggle =
-                                        this.props.multipleSelect ||
-                                        this.props.deselectEnabled;
-                                    this.checkOption(
-                                        i,
-                                        shouldToggle ? !choice.checked : true);
-                                }
-                            }}>
-                    <div>
-                        <span className="checkbox">
-                            <input
-                                ref={"radio" + i}
-                                type={inputType}
-                                name={radioGroupName}
-                                checked={choice.checked}
-                                onClick={(e) => {
-                                    // Avoid sending this to the parent
-                                    e.stopPropagation();
-                                }}
-                                onChange={(e) => {
-                                    this.checkOption(i, e.target.checked);
-                                }}
-                                disabled={this.props.apiOptions.readOnly} />
-                        </span>
-                        {/* A pseudo-label. <label> is slightly broken on iOS,
-                            so this works around that. Unfortunately, it is
-                            simplest to just work around that everywhere. */}
-                        <span
-                                className={
-                                    ApiClassNames.RADIO.OPTION_CONTENT + " " +
-                                    ApiClassNames.INTERACTIVE
-                                }
-                                style={{
-                                    cursor: "default",
-                                }}>
-                            {content}
-                        </span>
-                        {Exercises.cluesEnabled && this.props.showClues &&
-                            choice.checked &&
-                            <div className="perseus-radio-clue">
-                                {choice.clue}
-                            </div>}
-                    </div>
+                    }}>
+
+                    {React.createElement(( choice.isNoneOfTheAbove ?
```

Comment:
This one I'd rather keep it this way, because I aimed at having the none-of-above logic encapsulated in its own component.

In the example that you wrote for example you can see that you would have the `BaseRadio` component dealing with how the content and the class of the `Choice` element should be, depending if the choice is a standard or a none-of-above one.

But I don't think that that's a concern of `BaseRadio` which should just care of using a `Choice` or `ChoiceNoneAbove` component and let have them deal with those internal logic.

Of course we could then take that a step further and say that maybe `BaseRadio` shouldn't be bothered even to choose between `Choice` and `ChoiceNoneAbove` :smile:  but I think this as it is could be already good enough